### PR TITLE
Refactor KMT tasks

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR performs some refactoring of KMT tasks.

- Generate `vmconfig.json` file from a template and `platforms.json` file.
- Abstract functionality into `LibvirtDomain` and `CommandRunner` classes.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/system-probe/test/micro-vm-init.sh [hunk 1]
```diff
@@ -18,7 +18,12 @@ fi
 code=0
 /test-runner -retry "${retry_count}" -packages-run-config "${pkgs_run_config_file}" || code=$?
 
-cp /job_env.txt /ci-visibility/junit/
+if [ ! -f "/job_env.txt" ]; then
```
Comment (brycekahle) [inline test/new-e2e/system-probe/test/micro-vm-init.sh:21]: Is this for local running?

**Comment by brycekahle** [comment]
Do we need to handle non-`.xz` files anymore? Aren't they always going to have it?

File: test/new-e2e/system-probe/test/micro-vm-init.sh (hunk 1 above)
Comment (usamasaqib) [reply test/new-e2e/system-probe/test/micro-vm-init.sh]: Yes

**Comment by usamasaqib** [comment]
> Python linting needs fixing.
> 
> Looks like there is something wrong with your changes:
> 
> ```
> xxd: /home/kernel-version-testing/rootfs/jammy-server-cloudimg-x86_64.qcow2: No such file or directory
> xxd: /home/kernel-version-testing/rootfs/jammy-server-cloudimg-x86_64.qcow2: No such file or directory
> ```

Where is this from?

File: tasks/kernel_matrix_testing/vmconfig.py [hunk 2]
```diff
@@ -77,6 +78,9 @@
     "amzn_4.14": "amzn_4.14",
     "amzn_5.4": "amzn_5.4",
     "amzn_5.10": "amzn_5.10",
+    "amzn_3": "amzn_3",
+    "amazon_2023": "amzn_3",
+    "al3": "amzn_3",
```
Comment (brycekahle) [inline tasks/kernel_matrix_testing/vmconfig.py:83]: `amzn_3`? It is called Amazon Linux 2023 and not Amazon Linux 3

File: test/new-e2e/system-probe/config/platforms.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+{
+    "url_base": "https://dd-agent-omnibus.s3.amazonaws.com/kernel-version-testing/rootfs",
+    "x86_64": {
+        "ubuntu_16.04": "master/xenial-server-cloudimg-x86_64.qcow2.xz",
+        "ubuntu_18.04": "master/bionic-server-cloudimg-x86_64.qcow2.xz",
+        "ubuntu_20.04": "master/focal-server-cloudimg-x86_64.qcow2.xz",
+        "ubuntu_22.04": "master/jammy-server-cloudimg-x86_64.qcow2.xz",
+        "ubuntu_23.10": "master/mantic-server-cloudimg-x86_64.qcow2.xz",
```
Comment (brycekahle) [inline test/new-e2e/system-probe/config/platforms.json:8]: these should be updated to use the new path: `master/ubuntu-<version_num>-x86_64.qcow2.xz`

File: test/new-e2e/system-probe/config/platforms.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        "debian_11": "master/debian-11-generic-x86_64.qcow2.xz",
+        "debian_12": "master/debian-12-generic-x86_64.qcow2.xz",
+        "centos_79": "master/centos-7.9-x86_64.qcow2.xz",
+        "centos_8": "master/centos-8-x86_64.qcow2.xz"
+    },
+    "arm64": {
+        "ubuntu_18.04": "master/bionic-server-cloudimg-arm64.qcow2.xz",
+        "ubuntu_20.04": "master/focal-server-cloudimg-arm64.qcow2.xz",
+        "ubuntu_22.04": "master/jammy-server-cloudimg-arm64.qcow2.xz",
+        "ubuntu_23.10": "master/mantic-server-cloudimg-arm64.qcow2.xz",
```
Comment (brycekahle) [inline test/new-e2e/system-probe/config/platforms.json:24]: these should be updated to use the new path: master/ubuntu-<version_num>-arm64.qcow2.xz

File: tasks/kernel_matrix_testing/vmconfig.py (hunk 2 above)
Comment (usamasaqib) [reply tasks/kernel_matrix_testing/vmconfig.py:83]: changed to `amzn_2023`

**Review by brycekahle — APPROVED** [review]

**Pull request merged**